packages/components/src/mobile/bottom-sheet/stepper-cell/index.native.js

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * WordPress dependencies
+ */
+import { _x, sprintf } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Cell from '../cell';
+import Stepper from './stepper';
+import styles from './style.scss';
+
+const STEP_SPEED = 200;
+const DEFAULT_STEP = 1;
+
+class BottomSheetStepperCell extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.announceValue = this.announceValue.bind( this );
+		this.onDecrementValue = this.onDecrementValue.bind( this );
+		this.onDecrementValuePressIn = this.onDecrementValuePressIn.bind( this );
+		this.onIncrementValue = this.onIncrementValue.bind( this );
+		this.onIncrementValuePressIn = this.onIncrementValuePressIn.bind( this );
+		this.onPressOut = this.onPressOut.bind( this );
+		this.startPressInterval = this.startPressInterval.bind( this );
+	}
+
+	componentWillUnmount() {
+		clearTimeout( this.timeout );
+		clearInterval( this.interval );
+		clearTimeout( this.timeoutAnnounceValue );
+	}
+
+	onIncrementValue() {
+		const { step, maxValue, onChangeValue, value } = this.props;
+		const newValue = value + step;
+
+		if ( newValue <= maxValue ) {
+			onChangeValue( newValue );
+			this.announceValue( newValue );
+		}
+	}
+
+	onDecrementValue() {
+		const { step, minValue, onChangeValue, value } = this.props;
+		const newValue = value - step;
+
+		if ( newValue >= minValue ) {
+			onChangeValue( newValue );
+			this.announceValue( newValue );
+		}
+	}
+
+	onIncrementValuePressIn() {
+		this.onIncrementValue();
+		this.timeout = setTimeout( () => {
+			this.startPressInterval( this.onIncrementValue );
+		}, 500 );
+	}
+
+	onDecrementValuePressIn() {
+		this.onDecrementValue();
+		this.timeout = setTimeout( () => {
+			this.startPressInterval( this.onDecrementValue );
+		}, 500 );
+	}
+
+	onPressOut() {
+		clearTimeout( this.timeout );
+		clearInterval( this.interval );
+	}
+
+	startPressInterval( callback ) {
+		let counter = 0;
+		this.interval = setInterval( () => {
+			callback();
+			counter += 1;
+
+			if ( counter === 10 ) {
+				clearInterval( this.interval );
+				this.startPressInterval( callback, STEP_SPEED / 2 );
+			}
+		}, STEP_SPEED );
+	}
+
+	announceValue( value ) {
+		const { label } = this.props;
+
+		if ( Platform.OS === 'ios' ) { // On Android it triggers the accessibilityLabel with the value change
+			clearTimeout( this.timeoutAnnounceValue );
+			this.timeoutAnnounceValue = setTimeout( () => {
+				AccessibilityInfo.announceForAccessibility( `${ value } ${ label }` );
+			}, 300 );
+		}
+	}
+
+	render() {
+		const { label, icon, minValue, maxValue, value, separatorType } = this.props;
+		const isMinValue = value === minValue;
+		const isMaxValue = value === maxValue;
+
+		const accessibilityLabel = sprintf(
+			/* translators: accessibility text. Inform about current value. %1$s: Control label %2$s: Current value. */
+			_x( '%1$s. Current value is %2$s', 'Increase or decrement to change the value' ),
```

Comment:
Not sure if we should use `_x` here.

As far as I understand, `_x` is used when the same frase might need different translations depending on the context. We do have another instance of this same string, but not sure if the context is different (this is a picker while the other is the slider).

https://github.com/WordPress/gutenberg/blob/2ebfb64ce589f743b7caa8a8e1727b576755e16e/packages/components/src/mobile/bottom-sheet/range-cell.native.js#L152

If we keep it, I'd suggest to use a similar context string, something like: `Picker for selecting a number inside a range`.

Reply:
Oh right! we do have the Slider one. I'll change the `_x` thanks!!